### PR TITLE
chore: add total results indicator

### PR DESCRIPTION
**Pull request by almeidabbm**
Closes: #12301 

### Description:

https://github.com/user-attachments/assets/93af3a6d-801a-4348-9fb6-1b17d5e48d0f

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx [hunk 1]
```diff
@@ -87,6 +87,13 @@ export const MetricsTable = () => {
         fetchMoreOnBottomReached(tableContainerRef.current);
     }, [fetchMoreOnBottomReached]);
 
+    const totalResults = useMemo(() => {
+        if (!data) return 0;
+        // Return total results from the last page, this should be the same but still we want to have the latest value
+        const lastPage = data.pages[data.pages.length - 1];
+        return lastPage.pagination?.totalResults ?? 0;
+    }, [data]);
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:95]: isn't `flatData` sufficient? or is there more to consider I'm missing?

Comment (almeidabbm) [reply packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:95]: `flatData` will only include the results that have been fetched
`totalResults` on the other hand has the full count of the query when not paginated which is the total number of metrics

File: packages/frontend/src/features/metricsCatalog/components/MetricsTableTopToolbar.tsx [hunk 2]
```diff
@@ -164,36 +166,39 @@ const CategoriesFilter = () => {
...
+                                >
+                                    <MantineIcon icon={IconX} />
+                                </ActionIcon>
+                            )
+                        }
+                    />
+                    {/* Categories filter */}
+                    <CategoriesFilter />
+                </Group>
+                <Badge size="sm">{totalResults}</Badge>
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/MetricsTableTopToolbar.tsx:201]: can we make it `violet` as well?

**Review by joaoviana — APPROVED** [review]
one Q on the color!

File: packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:95]: ahhh I see! ty

**Pull request merged**